src/Sarif/IOptionsProvider.cs

```diff
@@ -7,6 +7,8 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public interface IOptionsProvider
     {
+        //GetOptions as a method may have significant processing and should not be an attribute
```

Comment:
"... should not be a property".

As it happens, the `SuppressMessageAttribute` class has a property named `Justification` that you can/should use instead of a comment here:

```
[System.Diagnostics.CodeAnalysis.SuppressMessage(
    "Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate",
    Justification = "This method is expensive and so should not be a property.")]
```